openassessment/staffgrader/models/submission_lock.py

```diff
@@ -0,0 +1,83 @@
+"""
+Models for locking Submissions for exclusive grading.
+Part of Enhanced Staff Grader (ESG).
+"""
+from django.db import models
+from django.utils.timezone import now
+from django.utils.translation import ugettext as _
+
+from openassessment.assessment.models.staff import StaffWorkflow
+from openassessment.staffgrader.errors.submission_lock import SubmissionLockContestedError
+
+
+class SubmissionGradingLock(models.Model):
+    """
+    Internal model for locking a submission for exclusive grading
+    """
+    TIMEOUT = StaffWorkflow.TIME_LIMIT
+
+    # NOTE - submission_uuid can refer to either the team or individual submission
+    submission_uuid = models.CharField(max_length=128, db_index=True, unique=True)
```

Comment:
I'd like to delay this until a separate PR until there's demonstrated contract for `StaffWorkflow`

Reply:
The "SubmissionList" view is really a StaffWorkflow view (all the data that we're looking for is on StaffWorkflow).

Reply:
The query currently doesn't specifically /need/ this FK but it's probably worth the performance

Reply:
I actually take that back, it would be a huge benefit. Rather than having to do complicated subqueries, I can just prefetch_related and hand it to the serializer

Reply:
If we add StaffWorkflow foreign key, any other constraints/info on how it's used?